yarl/_url.py

```diff
@@ -432,25 +438,28 @@ def __eq__(self, other: object) -> bool:
         if type(other) is not URL:
             return NotImplemented
 
-        val1 = self._val
-        scheme1, netloc1, path1, query1, fragment1 = val1
-        if not path1 and netloc1:
-            val1 = (scheme1, netloc1, "/", query1, fragment1)
-
-        val2 = other._val
-        scheme2, netloc2, path2, query2, fragment2 = val2
-        if not path2 and netloc2:
-            val2 = (scheme2, netloc2, "/", query2, fragment2)
-
-        return val1 == val2
+        path1 = "/" if not self._path and self._netloc else self._path
+        path2 = "/" if not other._path and other._netloc else other._path
+        return (
+            self._scheme,
+            self._netloc,
+            path1,
+            self._query,
+            self._fragment,
+        ) == (
+            other._scheme,
+            other._netloc,
+            path2,
+            other._query,
+            other._fragment,
+        )
```

Comment:
Could it be faster to use chained comparisons here (scheme == scheme and etc == etc) rather than build tuples?

Reply:
Let's try that

Reply:
Faster 👍